assets/javascripts/discourse/components/event-rsvp.js.es6

```diff
@@ -9,7 +9,7 @@ export default Ember.Component.extend({
 
   @computed('currentUser', 'topic.event_going')
   userGoing(user, eventGoing) {
-    return eventGoing && eventGoing.indexOf(user.username) > -1;
+    return eventGoing && eventGoing.indexOf(user.id.toString()) > -1;
```

Comment:
@fzngagan Could you briefly explain why it's necessary to cast this toString() here and below?

Reply:
@angusmcleod We're saving the rsvp users in a comma separated form. One way is to cast those into `int` on rails side.

I suggest we should save that list to a json array which would give us a ruby array on parsing and will preserve the actual data type. This is the pattern I suggest we should follow in all the similar cases. Thoughts?

Reply:
@fzngagan Yes, I can see the benefit in that. If you think this will be a relatively simple change (including migration), let's do it as part of this PR.

Reply:
Yes, this is not very difficult to migrate to IMO.

Reply:
@fzngagan I see you're now saving this as a JSON array :+1: To finish this off we need to cast the custom field to ``json`` using

```
Topic.register_custom_field_type('event_going', :json)
```
Then we won't need to handle parsing (it'll be handled in the ``has_custom_fields.rb`` concern).